offchain-modules/packages/x/src/reconc/CKBRecordObservable.ts

```diff
@@ -210,4 +225,116 @@ export class CKBRecordObservable {
+    ];
+    const committeeMultisigLockscript = parseAddress(getOmniLockMultisigAddress());
+    const searchKey: SearchKey = {
+      script_type: ScriptType.lock,
+      script: committeeMultisigLockscript,
+      filter: { block_range: blockRange },
+    };
+
+    const { rpc, indexer } = this.provider;
+
+    return from(indexer.get_transactions({ searchKey })).pipe(
+      expand((tx) => indexer.get_transactions({ searchKey, cursor: tx.last_cursor })),
+      takeWhile((tx) => tx.objects.length > 0),
+      mergeMap((tx) => tx.objects),
+      mergeMap(async (o) => {
+        const txWithStatus = await rpc.getTransaction(o.tx_hash);
+        return {
+          info: o,
+          tx: txWithStatus,
+        };
+      }),
+      takeWhile((ckbTxInfo) => {
+        const firstOutputLockscript = ckbTxInfo.tx.transaction.outputs[0].lock;
+        return (
+          firstOutputLockscript.codeHash === committeeMultisigLockscript.code_hash &&
+          firstOutputLockscript.hashType === committeeMultisigLockscript.hash_type &&
+          firstOutputLockscript.args === committeeMultisigLockscript.args
+        );
+      }),
+      mergeMap(async (ckbTxInfo) => {
+        // tx.objects.map((o) => o.tx_hash)
+        const transaction = ckbTxInfo.tx.transaction;
+        const lockTx = await parseLockTx(transaction);
+        const { amount, recipientAddress, assetIdent, senderAddress } = lockTx!;
+        return {
+          mintId: `${transaction.hash}-${ckbTxInfo.info.tx_index}`,
+          amount: `0x${amount.toString(16)}`,
+          txId: transaction.hash,
+          sender: senderAddress,
+          token: assetIdent,
+          recipient: recipientAddress,
+          sudtExtraData: '0x',
+          blockNumber: Number(ckbTxInfo.info.block_number),
+          blockHash: ckbTxInfo.tx.txStatus.blockHash!,
+        };
+      }),
+    );
+  }
+
+  observeUnlockRecord(filter: CKBUnlockFilter): Observable<CkbUnlockRecord> {
+    const blockRange: SearchKeyFilter['block_range'] = [
+      filter.fromBlock ? filter.fromBlock : '0x0',
+      filter.toBlock ? filter.toBlock : '0xffffffffffffffff', // u64::Max
+    ];
+    const committeeMultisigLockscript = parseAddress(getOmniLockMultisigAddress());
+    const searchKey: SearchKey = {
+      script_type: ScriptType.lock,
+      script: committeeMultisigLockscript,
+      filter: { block_range: blockRange },
+    };
+
+    const { rpc, indexer } = this.provider;
+
+    return from(indexer.get_transactions({ searchKey })).pipe(
+      expand((tx) => indexer.get_transactions({ searchKey, cursor: tx.last_cursor })),
+      takeWhile((tx) => tx.objects.length > 0),
+      mergeMap((tx) => tx.objects),
+      mergeMap(async (o) => {
+        const txWithStatus = await rpc.getTransaction(o.tx_hash);
+        return {
+          info: o,
+          tx: txWithStatus,
+        };
+      }),
+      rxFilter((ckbTxInfo) => {
+        const firstOutputLockscript = ckbTxInfo.tx.transaction.outputs[0].lock;
```

Comment:
inputs[0].lock === committeeLock for unlockTx

Reply:
getTransactions by committeeLock searchKey to get all lock txs and unlock txs, distinguish lock tx or unlock tx according to whether the first output is equal to committeeLock